lib/search.rb

```diff
@@ -353,15 +353,15 @@ def self.advanced_order(trigger, &block)
   end
 
   def self.advanced_orders
-    @advanced_orders
+    @advanced_orders || {}
   end
 
   def self.advanced_filter(trigger, &block)
     (@advanced_filters ||= {})[trigger] = block
   end
 
   def self.advanced_filters
-    @advanced_filters
+    @advanced_filters || {}
```

Comment:
Instead of having to check for `@advanced_filters` and return an empty hash if it’s not defined, we could probably set it here and then just access it normally in `.advanced_filter`:
```ruby
def self.advanced_filters
  @advanced_filters ||= {}
end

def self.advanced_filter(trigger, &block)
  advanced_filters[trigger] = block
end
```
We could do the same thing for `.advanced_orders`.

Reply:
Nice